src/main/scala/io/github/edefritz/commands/Tile38Command.scala

```diff
@@ -2,17 +2,152 @@ package io.github.edefritz.commands
+    args.add(key)
+    args.add(id)
+
+    fields.foreach { f =>
+      f.fields.foreach {
+        case (fieldName, fieldValue) => args.add(f.keyword).add(fieldName).add(fieldValue)
+      }
+    }
+
+    ex.foreach { e =>
+      args.add(e.keyword).add(e.seconds)
+    }
+
+    setCondition.foreach { n =>
+      args.add(n.keyword)
+    }
+
+    inputFormat match {
+      case SetPoint(lat, lon, z) =>
+        args
+          .add(inputFormat.keyword)
+          .add(lat)
+          .add(lon)
+        if (z.isDefined) args.add(z.get)
+      case SetObject(geojson) => args.add(inputFormat.keyword).add(geojson)
+      case SetBounds(minlat, minlon, maxlat, maxlon) =>
+        args.add(inputFormat.keyword).add(minlat).add(minlon).add(maxlat).add(maxlon)
+      case SetHash(hash)     => args.add(inputFormat.keyword).add(hash)
+      case SetString(string) => args.add(inputFormat.keyword).add(string)
+      case other             => args.add(other.keyword)
+    }
+    Success(args)
+  }
+}
+
+object SetCommand {
+  def apply(
+      key: String,
+      id: String,
+      inputFormat: SetCommandArgument with SetCommandInputFormat,
+      fields: Map[String, Double] = Map.empty,
+      ex: Int = 0,
+      setCondition: SetCommandArgument with SetCondition = Default
+  ): SetCommand = {
+    val fieldsArgument       = if (fields.nonEmpty) Some(Fields(fields)) else None
+    val exArgument           = if (ex > 0) Some(Ex(ex)) else None
+    val setConditionArgument = Some(setCondition)
+    this(key, id, inputFormat, fieldsArgument, exArgument, setConditionArgument)
+  }
+
+  sealed trait SetCommandArgument {
+    val keyword: String
+  }
+
+  sealed trait SetCondition
+
+  final case class NotExists() extends SetCommandArgument with SetCondition {
+    override val keyword: String = "NX"
+  }
+
+  /**
+    * Sets entry only if it already exists
+    */
+  final case class Exists() extends SetCommandArgument with SetCondition {
+    override val keyword: String = "XX"
+  }
+
+  // Adding default condition instead of option
+  final case class Default() extends SetCommandArgument with SetCondition {
+    override val keyword: String = ""
+  }
+
+  // Do we need this trait?
+  sealed trait SetCommandFields
+
+  final case class Fields(fields: Map[String, Double]) extends SetCommandArgument with SetCommandFields {
+    override val keyword: String = "FIELD"
+  }
+
+  final case class Ex(seconds: Int) extends SetCommandArgument {
+    override val keyword: String = "EX"
+  }
+
+  sealed trait SetCommandInputFormat
+
+  final case class Point(lat: Double, lon: Double, z: Option[Double] = None)
+      extends SetCommandArgument
+      with SetCommandInputFormat {
+    override val keyword: String = "POINT"
+  }
+
+  final case class Object(geojson: String) extends SetCommandArgument with SetCommandInputFormat {
+    override val keyword: String = "OBJECT"
+  }
+
+  final case class Bounds(minlat: Double, minlon: Double, maxlat: Double, maxlon: Double)
```

Comment:
To make this a bit more robust we could define specific fields for Lat / Long with https://github.com/fthomas/refined